Examples/Logicals/logical-example.fsh

```diff
@@ -0,0 +1,49 @@
+// @Name: Logical Example
+// @Description: An example of Logical resources and how to use them
+
+// A 'Logical' is neither a resource nor a datatype. It can be thought of
+// as a data class.
+Logical: AnimalAbilities
+* hasLungs 0..1 boolean "Does the animal have lungs?"
+* canSwim 0..1 boolean "Can the animal swim?"
+
+// A Logical can be used in place of a "complex datatypes"
+Logical: Animal
+* name 1..1 string "The animal's name"
+* age 0..1 decimal "The animal's age in years"
+* traits 1..1 AnimalAbilities "The animal's abilities"
+
+// A Logical can be a Parent (Inheritance)
+Logical: Pet
+Parent: Animal
+* isVaccinated 0..1 boolean "Is the pet vaccinated?"
+
+// A Logical can be like a Resource, but you have to Add an id-field and the 
+// #can-be-target characteristic, so that References work.
```

Comment:
Logicals that have [Element](https://hl7.org/fhir/R4/element.html) as a parent will inherit the `id` and `extension` elements. I think it might be worth noting that here or _somewhere_, as logicals usually have either `Base` or `Element` as the root parent (but SUSHI uses `Base` by default).

Reply:
Question for my understanding: When writing a data model, I tend end up with 
- things that are like FHIR Resources (these typically get their own REST endpoints)
- things that are like complex FHIR datatypes (these are typically contained inside several of my resources)

What would be the most appropriate Parent for the Resource-like things? I would like to use `Resource`, but this is not allowed. Using `Element` would give me the `id`-element, but it feels wrong, because `Element` is used for `DataType`s in FHIR...

Reply:
That's a good question. According to FHIR spec ([here](https://hl7.org/fhir/R5/logical.html#2.1.14)), parents can be `Base` or `Element` or an absolute URI for another logical model. I hadn't thought about Element being restricted to data types in FHIR, but you're absolutely right; the [documentation for Element](https://hl7.org/fhir/R5/types.html#Element) states that it is for "elements included in a resource" (which I guess makes sense). So perhaps it would not be appropriate to use it as a parent for Resource-like logicals, even if it conveniently has the `id` and `extension` already.

That's kind of a lot to explain in an example, so maybe it is better just to leave your example as-is?

Reply:
I'm absolutely open to further changes to this PR (or that someone with more experience takes it over) -- but I'm also OK with merging it as is and adapting it later.